app/src/main/java/zapsolutions/zap/fragments/AdvancedSettingsFragment.java

```diff
@@ -24,6 +27,19 @@ public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
 
         mUG = new UserGuardian(getActivity(), this);
 
+        // On change block explorer option
+        mSwBlockExplorer = findPreference("blockExplorer");
+        mSwBlockExplorer.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
+            @Override
+            public boolean onPreferenceChange(Preference preference, Object newValue) {
+                if (newValue != null && newValue.toString().equalsIgnoreCase("Blockstream (v3 Tor)")) {
+                    Toast.makeText(getActivity(), R.string.settings_blockExplorer_tor_toast, Toast.LENGTH_LONG).show();
+                    return true;
+                } else {
+                    return true;
+                }
```

Comment:
This could be simplified to:

If(new value...) {
Toast...
}
rmeturn true;

Reply:
copy. good catch. resolved